src/Servers/Kestrel/Core/test/StartLineTests.cs

```diff
@@ -516,6 +516,27 @@ public void AuthorityForms(string rawTarget, string path, string query)
         DifferentFormsWorkTogether();
     }
 
+    [Theory]
+    [InlineData("\r")]
+    [InlineData("\n")]
+    [InlineData("\r\r\r\r\r")]
+    [InlineData("\r\n")]
+    [InlineData("\n\r")]
+    [InlineData("\r\n\n")]
+    public void LeadingCrLfAreAllowed(string startOfRequestLine)
+    {
+        var rawTarget = "http://localhost/path1?q=123&w=xyzw";
+        Http1Connection.Reset();
+        // RawTarget, Path, QueryString are null after reset
+        Assert.Null(Http1Connection.RawTarget);
+        Assert.Null(Http1Connection.Path);
+        Assert.Null(Http1Connection.QueryString);
+
+        var ros = new ReadOnlySequence<byte>(Encoding.ASCII.GetBytes($"{startOfRequestLine}CONNECT {rawTarget} HTTP/1.1\r\n"));
```

Comment:
Only connect?

Reply:
The method doesn't matter for this today but I suppose we could somehow change that in the future :smile:. I'll vary the method.

Reply:
I know I guess I'd like to see it added for all the verbs we hardcode and then one unknown